src/unix/loop.c

```diff
@@ -68,14 +68,17 @@ int uv_loop_init(uv_loop_t* loop) {
   loop->child_watcher.flags |= UV__HANDLE_INTERNAL;
   QUEUE_INIT(&loop->process_handles);
 
-  if (uv_rwlock_init(&loop->cloexec_lock))
-    abort();
+  err = uv_rwlock_init(&loop->cloexec_lock);
+  if (err)
+    return err;
 
-  if (uv_mutex_init(&loop->wq_mutex))
-    abort();
+  err= uv_mutex_init(&loop->wq_mutex);
```

Comment:
If this fails you need to destroy the rwmutex that was created before, or we'd leak resources.

Same goes for every other resource. If uv_loop_init returns error, everything should have been destroyed, since user shouldn't have to call uv_loop_close.